src/Kentico.Xperience.Lucene.Core/Scaling/RebuildWebFarmTask.cs

```diff
@@ -0,0 +1,29 @@
+﻿using CMS.Base;
+using CMS.Core;
+
+using Kentico.Xperience.Lucene.Core.Indexing;
+
+namespace Kentico.Xperience.Lucene.Core.Scaling;
+internal class RebuildWebFarmTask : WebFarmTaskBase
+{
+    private readonly IEventLogService eventLog;
+    private readonly ILuceneClient luceneClient;
+    public string? IndexName { get; set; }
+    public string? CreatorName { get; set; }
+
+    public RebuildWebFarmTask()
+    {
+        eventLog = Service.Resolve<IEventLogService>();
+        luceneClient = Service.Resolve<ILuceneClient>();
+    }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+    public override async void ExecuteTask()
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+    {
+        string message = $"Server {SystemContext.ServerName} is processing a Lucene Rebuild task from creator {CreatorName}";
+        eventLog.LogInformation("Lucene Rebuild Task", "Execute", message);
+
+        luceneClient.Rebuild(IndexName!, default).GetAwaiter().GetResult();
```

Comment:
```suggestion
    public override async Task ExecuteTask()
    {
        string message = $"Server {SystemContext.ServerName} is processing a Lucene Rebuild task from creator {CreatorName}";
        eventLog.LogInformation("Lucene Rebuild Task", "Execute", message);

       await luceneClient.Rebuild(IndexName!, default);
```

I don't think `async void` is a good choice. Either the method should be synchronous and call `.GetAwaiter().GetResult()` on the `Task` that `Rebuild()` returns or it should be asynchronous and return a `Task` to the caller.

Reply:
I deleted the `async void`. The `WebFarmTaskBase` does not provide a method with `Task` return type.